### PR TITLE
Introduce a cell model based on Thevenin

### PR DESCRIPTION
**Pull request by WardLT**
Use the [Thevenin](https://github.com/ROVI-org/thevenin) to implement an ECM model.

To do list:
- [x] Gather overpotentials from RC elements
- [x] Make sure it works with a state estimator
- [x] Review implementation with @c-randall 
- [x] Document how to make new CellModels, start a list

### CODE REVIEW

File: moirae/models/thevenin/__init__.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                params[scalar] = value[b % value.shape[0], 0]
+
+            # Add the SOC and series resistors as functions where we pin the batch ID to the appropriate value
+            params['ocv'] = partial(asoh.ocv, batch_id=b)
+            params['R0'] = partial(asoh.r[0], batch_id=b)
+
+            # Append the RC elements
+            for r in range(params['num_RC_pairs']):
+                params[f'R{r + 1}'] = partial(asoh.r[r + 1], batch_id=b)
+                params[f'C{r + 1}'] = partial(asoh.r[r], batch_id=b)
```
[the remark this answers is not in the document]
Comment (WardLT) [reply moirae/models/thevenin/__init__.py:55]: Interesting. Did you make your `lambda soc, ...` function inside a for loop that also involved making `r_copy`? If so, the value of `r_copy` will be the last one created in the loop because the code inside a lambda is evaluated [when the code is called](https://docs.python.org/3/glossary.html#term-lambda):

```
>>> f = lambda x: a + 2  # a is not defined, but that's OK
>>> a = 2  
>>> f(1)  # Because the value isn't evaluated until here
4
>>>
```

Partial functions, on the other hand, hold on to the object that was passed during creation (so it must exist)

```
>>> a = 4
>>> f = partial(np.add, a)
>>> a = 2
>>> f(2)
6
```

As an aside, the above example is a good reason to avoid assigning Lambda functions to a variable. They're really only meant to be temporary, as also visible in that Python won't serialize them.

TL;DR: I think `lambda` is the problem, but am going to add a test anyway

[the remark this answers is not in the document]
Comment (victorventuri) [reply moirae/models/thevenin/__init__.py:55]: Ah, I see, I think your hunch may be correct, as I was defining the `lambda`s inside a `for`-loop. At any rate, when writing the tests, I would encourage us to test a case with more than one RC element in the ECM.

[the remark this answers is not in the document]
Comment (WardLT) [reply moirae/models/thevenin/__init__.py]: Done. Caught that I was using the resistance for both R and C while at it 🤦🏼‍♂️

**Comment by WardLT** [comment]
@c-randall , this is all ready. Not sure if you have bandwidth or desire to review, but want to let you know the first iteration is complete

File: moirae/models/thevenin/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (victorventuri) [reply moirae/models/thevenin/__init__.py]: sweet! code twice, test thrice 💡

**Review by victorventuri — APPROVED** [review]
Nice work, tests seem very thorough too!
